server/db/posts.js

```diff
@@ -1,6 +1,21 @@
 import { resolveInclude } from 'ejs';
 import {db} from './database.js';
 
+export async function findByUsername(username) {
+    return db.execute('SELECT * FROM posts WHERE username=?', [username])
+    .then(result => result[0]);
+}
+
+export async function countByUsername(username) {
+    return db.execute('SELECT COUNT(*) as count FROM posts WHERE username=?', [username])
+    .then(result => result[0][0].count);
+}
+
+export async function findByPostId(id) {
+    return db.execute('SELECT * FROM posts WHERE id=?', [id])
+    .then(result => result[0][0]);
```

Comment:
위 코드에서
```javascript
.then(result => result[0]);
```
```javascript
.then(result => result[0][0].count);
```
```javascript
.then(result => result[0][0]);
```
는 무슨 의미인가요? (count는 알겠지만..)
조회한 데이터를 가공해서 반환하고 싶은 거라면, 어떤 데이터를 반환하고자 하는건지 코드에 나타났으면 좋겠습니다.
코드만 봤을때는 findByUsername의 result[0]와 findByPostId의 result[0][0]가 뭘 반환하는지 모르겠거든요.

Reply:
주석을 달면 될까요..? 어떤식으로 나타내는게 좋을지 모르겠습니다..!
`result`를 출력해보면 `[BinaryRow배열, ColumnDefinition배열]`이 나와서, `result[0]`으로 가져오려는 내용에 해당하는 부분인 `BinaryRow배열`을 리턴해주었고, `result[0][0]`으로 배열의 첫번째 값을 리턴했습니다. 

![image](https://user-images.githubusercontent.com/61776419/130004853-f5d586ab-c37d-4b4b-9610-9cd86a967fd0.png)
사진은 `findByPostId(id)`에서 `result[0]`일 때 반환하는 `BinaryRow배열`입니다.

Reply:
mysql2 [공식문서](https://www.npmjs.com/package/mysql2)를 보면 이렇게 결과를 받고 있네요.
```javascript
const [rows, fields] = await connection.execute('SELECT * FROM `table` WHERE `name` = ? AND `age` > ?', ['Morty', 14]);
```
이렇게도 받을 수 있군요.
```javascript
connection.execute(
  'SELECT * FROM `table` WHERE `name` = ? AND `age` > ?',
  ['Rick C-137', 53],
  function(err, results, fields) {
    console.log(results); // results contains rows returned by server
    console.log(fields); // fields contains extra meta data about results, if available

    // If you execute same statement again, it will be picked from a LRU cache
    // which will save query preparation time and give better performance
  }
);
```
`.then`을 사용하려면 이렇게도 받을 수 있겠네요.
```javascript
con.promise().query("SELECT 1")
  .then( ([rows,fields]) => {
    console.log(rows);
  })
  .catch(console.log)
  .then( () => con.end());
```

Reply:
감사합니다. 참고해서 수정했습니다!